include/boost/math/distributions/hyperexponential.hpp

```diff
@@ -282,12 +277,13 @@ class hyperexponential_distribution
     // Two arg constructor from 2 ranges, we SFINAE this out of existence if
     // either argument type is incrementable as in that case the type is
     // probably an iterator:
-    public: template <typename ProbRangeT, typename RateRangeT>
+    public: template <typename ProbRangeT, typename RateRangeT, 
```

Comment:
Sorry Matt, but this is going to be a lot more complex: we have two sets of 2-arg constructors, one accepts 2 ranges, conceptually that's anything that has begin() and end() member functions, but now that C++20 is here, we should probably take the opportunity to check that we're compatible with native range concepts.  The other constructor accepts 2 iterators.  So the use of has_pre_incremement is a simplistic way to check for an iterator concept, and I'm afraid is_pointer doesn't match this.

BTW the "modern" way to implement concept-like-traits is via is_detected: https://www.boost.org/doc/libs/1_75_0/libs/type_traits/doc/html/boost_typetraits/reference/is_detected.html  GCC has it's own version but it's not in the std sadly so other compilers do not :(

Reply:
I saw the comment about using a pointer and was hopeful that would work. `std::iterator_traits` is only SFINAE friendly in C++14 and on or that would be the easy button. I'll have to take a look at your link and come back with a more robust solution.